src/services/firebase.js

```
@@ -47,13 +59,80 @@ export async function loginAnonymously(displayName = '') {
+    console.error('Password reset error:', error);
+    throw error;
+  }
+}
+
+// Function to convert anonymous account to permanent account
+export async function convertAnonymousAccount(email, password) {
+  try {
+    const auth = getAuth();
+    const user = auth.currentUser;
+
+    if (user && user.isAnonymous) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)